File: python/01.py
filename = '01.txt'

value = 0
position = 1

with open(filename) as f:
    while True:
        c = f.read(1)
        if not c:
            break
        if c == "(":
            value = value + 1
        if c == ")":
            value = value - 1
        if value < 0:
            print "position is", position
            break
        position = position + 1

print "value is", value

File: javascript/01.js
var fs = require('fs');

var filename = '01.txt';
var contents = fs.readFileSync(filename).toString().trim();

function countFloors(floors) {
  var symbols = floors.split('');
  console.log(symbols);
  var floorCount = symbols.map(s => {
                                      switch(s) {
                                        case '(': return 1;
                                        case ')': return -1;
                                        default: return;
                                      }
                                    })
                                    .reduce((previous, current) => previous + current);
  return floorCount;
}

console.log(countFloors(contents));
